src/handlers/transfers/handler.js

```diff
@@ -0,0 +1,318 @@
+
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ * Lazola Lucas <lazola.lucas@modusbox.com>
+ * Rajiv Mothilal <rajiv.mothilal@modusbox.com>
+ * Miguel de Barros <miguel.debarros@modusbox.com>
+ * Deon Botha <deon.botha@modusbox.com>
+
+ --------------
+ ******/
+'use strict'
+
+const Logger = require('@mojaloop/central-services-shared').Logger
+const TransferHandler = require('../../domain/transfer')
+const Utility = require('../lib/utility')
+const DAO = require('../lib/dao')
+const Kafka = require('../lib/kafka')
+const Validator = require('./validator')
+const TransferQueries = require('../../domain/transfer/queries')
+
+const TRANSFER = 'transfer'
+const PREPARE = 'prepare'
+const FULFIL = 'fulfil'
+const REJECT = 'reject'
+
+/**
+ * @method prepare
+ *
+ * @async
+ * This is the consumer callback function that gets registered to a topic. This then gets a list of message,
+ * we will only ever use the first message in non batch processing. We then break down the message into its payload and
+ * begin validating the payload. Once the payload is validated successfully it will be written to the database to
+ * the relevant tables. If the validation fails it is still written to the database for auditing purposes but with an
+ * ABORT status
+ *
+ * @param {error} error - error thrown if something fails within Kafka
+ * @param {array} messages - a list of messages to consume for the relevant topic
+ *
+ * @function Validator.validateByName to validate the payload of the message
+ * @function TransferQueries.getById checks if the transfer currently exists
+ * @function TransferHandler.prepare creates new entries in transfer tables for successful prepare transfer
+ * @function TransferHandler.reject rejects an existing transfer that has been retried and fails validation
+ *
+ * @returns {object} - Returns a boolean: true if successful, or throws and error if failed
+ */
+const prepare = async (error, messages) => {
+  if (error) {
+    // Logger.error(error)
+    throw new Error()
+  }
+  let message = {}
+  try {
+    if (Array.isArray(messages)) {
+      message = messages[0]
+    } else {
+      message = messages
+    }
+    Logger.info('TransferHandler::prepare')
```

Comment:
@wdcmb @NicoDuvenage @ksatya77 @elnyry : See the code below. This is exactly why we need a Rules engine. The section below is going to end up being a tree of entangled hard to read and maintain code. :(

Reply:
Could not agree more.

Reply:
so what should i change it to? Any suggestions

Reply:
Nothing for now. Will need to refactor this once we have a rules engine.